pkgs/development/interpreters/python/tests.nix

```diff
@@ -91,11 +122,21 @@ let
     testfun = name: attrs: runCommand "${python.name}-tests-${name}" ({
       inherit (python) pythonVersion;
     } // attrs) ''
+      mkdir $out
+
+      # set up the test files
       cp -r ${./tests/test_environments} tests
       chmod -R +w tests
       substituteAllInPlace tests/test_python.py
-      ${attrs.interpreter} -m unittest discover --verbose tests #/test_python.py
-      mkdir $out
+
+      # run the tests by invoking the interpreter directly
+      ${attrs.interpreter} -m unittest discover --verbose tests | tee "$out/direct.txt"
+
+      # run the tests by invoking the interpreter via $PATH
+      export PATH="$(dirname ${attrs.interpreter}):$PATH"
+      "$(basename ${attrs.interpreter})" -m unittest discover --verbose tests | tee "$out/path.txt"
```

Comment:
This tests the scenario that @3541 discovered - invoking the python interpreter via $PATH.